### PR TITLE
Support Friendly URLs

### LINKED ISSUE #336: Support for Friendly/Semantic URLs in Viewer
As a user of the OSCAL viewer app, I want to be able use bookmarkable, friendly/semantic URLs when the app is configured for REST mode so that I can share and bookmark specific OSCAL URLs to be able to quickly view those artifacts without additional navigation steps.

### Acceptance Criteria
- When the app is [configured for REST mode](https://github.com/EasyDynamics/easygrc/issues/6) a URL with the OSCAL object type and UUID can be used to load the corresponding artifact
  - Example: https://oscal-viewer.msd.easydynamics.com/system-security-plan/cff8385f-108e-40a5-8f7a-82f3dc0eaba8
  - A URL path element for the UUID is preferred but a URL parameter would be acceptable

### PR DESCRIPTION
**Pull request by tuckerzp**
Rather than loading by OscalURL's in the editor, we now load based on
the OSCALObjects uuid. This allows for a lot more coherent way of
routing throughout the editor. It also allows you to load an element by
placing the uuid in the url.

This may render some portions of the `OSCALLoader` redundant. As I still do not
feel completely confident on what everything in the `OSCALLoader` does, please
pay extra attention in case I missed something. 

Testing still needs to be added to this, but the functionality should 
be working. It makes more sense to deal with this testing in https://github.com/EasyDynamics/oscal-editor-deployment/issues/98

Closes #336

### CODE REVIEW

**Review by kylelaker — COMMENTED** [review]
On the whole... wow! Love it! I think there are some small adjustments we can make to reduce duplication further.

File: src/components/OSCALLoader.js [hunk 1]
```diff
@@ -132,8 +133,8 @@ export default function OSCALLoader(props) {
     setOscalUrl(event.target.value);
   };
 
-  const handleUuidChange = (event) => {
-    const newOscalUrl = `${props.backendUrl}/${props.oscalObjectType.restPath}/${event.target.value}`;
+  const handleUuidChange = (objectUuid) => {
+    const newOscalUrl = `${props.backendUrl}/${props.oscalObjectType.restPath}/${objectUuid}`;
```
Comment (kylelaker) [inline src/components/OSCALLoader.js:137]: This breaks the API expected by the `OSCALLoaderForm` props correct? See Line 218?

Comment (tuckerzp) [reply src/components/OSCALLoader.js:137]: I may misunderstand what you mean, but `OSCALLoaderForm` was changed to deal with this.

https://github.com/EasyDynamics/oscal-react-library/blob/eb50a0174471509cd83c1068b8a7d0f12f5b3a42/src/components/OSCALLoaderForm.js#L113

`event.target.value` is the `uuid` here.

**Comment by kylelaker** [comment]
Okay so it looks like there is an odd behavior here.

### Steps to Reproduce

1. Launch the editor, go to any OSCAL Document type (for this example, Catalog)
2. Choose a catalog from the drop down
3. Wait for the catalog to load (or don't)
4. Go to the navigation
5. Click the same OSCAL document type (in this example, Catalog)

### Expected Behavior

The page is blank except for the LoaderForm.

### Actual Behavior

The previous document remains on the page.

**Comment by tuckerzp** [comment]
> Expected Behavior
The page is blank except for the LoaderForm.

> Actual Behavior
The previous document remains on the page.

I agree that it doing that would make sense but I am not 100% sure this is an issue that needs to be solved here. #472 will get rid of the ability to click on a `OSCALType` like that. So going to just `catalog` will no longer be possible. Making this behavior impossible to replicate.

With that said, I also think that what we push to develop should be a working version of the editor. But it seems that `Develop` also has this same behavior. Although this branch does remove the `OSCALLoaderForms` data.

I am fully open to working on fixing this, I just wanted to be sure that it is something worth fixing on this pr.

File: example/src/App.js [hunk 2]
```diff
@@ -89,6 +88,80 @@ function App() {
...
+      loaderElement: OSCALSSPLoader,
+    },
+  ];
+
+  const appBarRoutes = (
+    <Route path="/">
+      <Route index element={`OSCAL Catalog ${appType}`} />
+      {oscalObjectTypes.map((oscalObjectType) => (
+        <Route path={oscalObjectType.pathName} key={oscalObjectType.pathName}>
+          {["", ":id"].map((path) => (
```
Comment (kylelaker) [inline example/src/App.js:119]: Because we use `["", ":id"]` twice, can we extract it to a variable and then use that in each place?

**Review by kylelaker — APPROVED** [review]

**Pull request merged**